Dip/Dip/Definition.swift

```diff
@@ -50,29 +50,70 @@ func ==(lhs: DefinitionKey, rhs: DefinitionKey) -> Bool {
 public enum ComponentScope {
   /// Indicates that a new instance of the component will be created each time it's resolved.
   case Prototype
+  /// Indicates that instances will be reused during resolve but will be discurded when topmost `resolve` method returns.
+  case ObjectGraph
   /// Indicates that resolved component should be retained by container and always reused.
   case Singleton
 }
 
 ///Definition of type T describes how instances of this type should be created when they are resolved by container.
```

Comment:
Now that this has two generic parameters, maybe better describe them in the comments, like "`T` is the type of the instance to resolve to, `F` is the type of the factory to generate that instance, namely the runtime parameters that needs to be given when resolving that instance" or sthg similar.